lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def with_context(context_key, options = {}, &block)
+    if !config.contexts.key?(context_key)
+      raise ArgumentError, "#{context_key} not found in Retriable.config.contexts"
```

Comment:
you could maybe make this exception more useful by listing `#{Retriable.config.keys}` /shrug